eds/plugins/vcs_provider_github.py

```
@@ -0,0 +1,156 @@
+        """
+        self._yaml = yaml
+        self._validate_schema()
+        self.validate()
+        self.overridden = False
+
+        token_env_var = self._yaml.get('token_env_var', 'GITHUB_TOKEN')
+        github_enterprise_url = self._yaml.get('github_enterprise_url', '')
+
+        # Login to Github or Github Enterprise.
+        token: str = os.environ.get(token_env_var)
+        if token is None or token == '':
+            raise RuntimeError(
+                f"ERROR: You must export the {token_env_var} environment variable."
+            )
+        if github_enterprise_url is None or github_enterprise_url == '':
+            self._g: GitHub = login(token=token)
+        else:
+            self._g: GitHubEnterprise = enterprise_login(url=github_enterprise_url, token=token)
+
+    def parse_event(self, payload: str) -> Dict:
+        """Parse webhook event for project url and ref."""
+        # parse project url and sha from the webhook payload.
+        json_data = json.loads(payload)
+        project_url = json_data['repository']['url']
+        head_sha = json_data['check_suite']['head_sha']
+        return {"url": project_url, "ref": head_sha}
+
+    def get_files(self, owner: str, repo_name: str, path: str = '/', ref: str = 'master') -> Dict[str, Contents]:
+        """Get project files.
+
+        Return the contents of the repository's specified ``ref``, under the
+        specified path, as a dict of string key to string file contents.
+
+        :param owner: owner of the repo
+        :param repo_name: name of the repo
+        :param path: path under the repo to get
+        :param ref: ref to get contents at
+        :return: repository contents
+        """
+        try:
+            repo: Repository = self._g.repository(owner, repo_name)
+            contents: Dict[str, Contents] = repo.directory_contents(path, ref=ref, return_as=dict)
+        except Exception as ex:
+            print(f"Exception in get_files: {ex}")
```

This print statement should at least change to a logging statement... but I'm still not sure if ignoring the exception is what we want?